var/spack/repos/builtin/packages/podio/package.py

```diff
@@ -29,6 +29,11 @@ class Podio(CMakePackage):
     version('0.9', sha256='3cde67556b6b76fd2d004adfaa3b3b6173a110c0c209792bfdb5f9353e21076f')
     version('0.8', sha256='9d035a7f5ebfae5279a17405003206853271af692f762e2bac8e73825f2af327')
 
+    variant('cxxstd',
+            default='17',
+            values=('17', '20'),
```

Comment:
Would this:
```suggestion
            values=('17', conditional('20', when='@0.14.1:')),
```
work?

Reply:
Apparently this does not work. Should it?

With those changes and removed conflict clause below, I get the following concretization:
```console
12:06:16 wdconinc@menelaos ~/git/spack (podio-cxxstd *$%=) $ spack spec -I podio@0.14 cxxstd=20
Input spec
--------------------------------
 -   podio@0.14 cxxstd=20

Concretized
--------------------------------
 -   podio@0.14%gcc@12.0.1~ipo~sio build_type=RelWithDebInfo cxxstd=20 patches=ee42196,fd9c373 arch=linux-ubuntu22.04-skylake
[+]      ^cmake@3.23.1%gcc@12.0.1~doc+ncurses+ownlibs~qt build_type=Release arch=linux-ubuntu22.04-skylake
[+]          ^ncurses@6.2%gcc@12.0.1~symlinks+termlib abi=none arch=linux-ubuntu22.04-skylake
... etc
```

Reply:
The issue here is with version:
```
>>> s = spack.spec.Spec('@0.14.1:')
>>> s.satisfies('@0.14')
True
```
The construct above should work if you e.g. use `0.13`

Reply:
Resolved with 002dceb0c3.